spec/lib/utils/metrics/snmp-txrx-counters-spec.js

```diff
@@ -0,0 +1,105 @@
+// Copyright 2016, EMC, Inc.
+/* jshint node: true */
+
+'use strict';
+
+describe("SNMP TxRx Counters Metric", function () {
+    var waterline;
+    var SnmpTxRxCountersMetric;
+    var metric;
+
+    before('SNMP TxRx Counters Metric before', function () {
+        helper.setupInjector([
+            helper.require('/lib/utils/metrics/base-metric.js'),
+            helper.require('/lib/utils/metrics/snmp-txrx-counters.js'),
+            helper.require('/lib/utils/job-utils/net-snmp-tool.js'),
+            helper.require('/lib/utils/job-utils/net-snmp-parser.js')
+        ]);
+
+        SnmpTxRxCountersMetric =
+            helper.injector.get('JobUtils.Metrics.Snmp.TxRxCountersMetric');
+        waterline = helper.injector.get('Services.Waterline');
+    });
+
+    beforeEach('SNMP TxRx Counters Metric beforeEach', function() {
+        metric = new SnmpTxRxCountersMetric('testnodeid', 'testhost', 'testcommunity');
+        waterline.catalogs = {
+            findMostRecent: sinon.stub().resolves()
+        };
+    });
+
+    it('should have a collectCounterMetricData function', function() {
+        expect(metric).to.have.property('collectCounterMetricData').that.is.a('function');
+    });
+
+    it('should collect counters data', function() {
+        metric.snmptool = { collectHostSnmp: sinon.stub() };
+        var values = { testoid1: '1', testoid2: '2' };
+        metric.snmptool.collectHostSnmp.resolves([ { values: values } ]);
+
+        return expect(metric.collectCounterData()).to.eventually.deep.equal(values);
+    });
+
+    it('should calculate counter data from snmp data', function() {
+        metric.snmptool = { collectHostSnmp: sinon.stub() };
+        metric.snmptool.collectHostSnmp.resolves([
+            { values: { '0': 'TestInterface' } }
+        ]);
+
+        return metric.updateOidDescriptionMapByType('names')
+            .then(function() {
+                var values = {
+                    'IF-MIB::ifInOctets.0': 0,
```

Comment:
All identical value `0` is not good for unit-testing, please change to some different number